app/src/main/kotlin/compiler/semantic_analysis/NameResolver.kt

```diff
@@ -0,0 +1,12 @@
+package compiler.semantic_analysis
+
+import compiler.ast.AstNodeTypes
+import compiler.ast.Program
+import compiler.common.diagnostics.Diagnostics
+import compiler.common.semantic_analysis.ReferenceHashMap
+
+object NameResolver {
+    fun calculateNameResolution(ast: Program, diagnostics: Diagnostics): ReferenceHashMap<Any, AstNodeTypes.NamedNode> {
```

Comment:
This can return just `ReferenceMap` (the same is with the other `calculate` functions).